formats/log/note.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"fmt"
+
+	"golang.org/x/mod/sumdb/note"
+)
+
+// CheckpointParser is responsible for parsing checkpoints for a specific log.
+// Any valid note must have been signed by all verifiers known by this parser.
+type CheckpointParser struct {
+	logVerifier       note.Verifier
+	logID             string
+	otherVerifiers    note.Verifiers
+	requiredOtherSigs int
+}
+
+// NewCheckpointParser creates a parser for checkpoints that must be signed by the log
+// which is verified by logVKey, and *all* of the other verifiers under otherVerifiers.
+func NewCheckpointParser(logVKey string, logID string, otherVerifiers ...string) (CheckpointParser, error) {
+	logVerifier, err := note.NewVerifier(string(logVKey))
+	if err != nil {
+		return CheckpointParser{}, fmt.Errorf("NewVerifier(%s): %v", logVKey, err)
+	}
+
+	ovs := make([]note.Verifier, len(otherVerifiers))
+	for i, k := range otherVerifiers {
+		ovs[i], err = note.NewVerifier(k)
+		if err != nil {
+			return CheckpointParser{}, fmt.Errorf("NewVerifier(%s): %v", k, err)
+		}
+	}
+	return CheckpointParser{
+		logVerifier:       logVerifier,
+		logID:             logID,
+		otherVerifiers:    note.VerifierList(ovs...),
+		requiredOtherSigs: len(otherVerifiers),
+	}, nil
+}
+
+// Parse returns a checkpoint parsed from the raw note, and any error if the note could
+// not be parsed, or any of the required signatures were missing.
+func (p CheckpointParser) Parse(chkptRaw []byte) (Checkpoint, *note.Note, error) {
```

Comment:
If I convinced you about the verifiers above, then maybe this could become `log.ParseCheckpoint(cpRaw, logID, logSigV, otherSigV, numOtherSigsRequired) (Checkpoint, *note.Note, error)`?

Reply:
I've met you half way. The number of signatures required is left to the caller to verify.